### PR TITLE
fix: ApplicationSet e2e TestSyncPolicyCreateUpdate&TestSyncPolicyCreateOnly test race condition bug #19577

### PR DESCRIPTION
**Pull request by a244591052**
Fixes [ISSUE #19577 ]

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by wanghong230 — APPROVED** [review]
Thank you for detailed explannation.

**Comment by blakepettersson** [comment]
I can confirm that these tests have been super-racy, thanks for the fix!

**Review by blakepettersson — APPROVED** [review]

File: test/e2e/applicationset_test.go [hunk 1]
```diff
@@ -915,7 +918,7 @@ func TestSyncPolicyCreateUpdate(t *testing.T) {
 
 		// Delete the ApplicationSet, and verify it deletes the Applications
```
Comment (pasha-codefresh) [inline test/e2e/applicationset_test.go:919]: could you please change the comment, probably better explain why apps still exists

**Review by pasha-codefresh — DISMISSED** [review]

**Review by christianh814 — APPROVED** [review]

**Comment by christianh814** [comment]
LGTM once Pasha's request is addressed 👍🏽

**Review by rbreeze — APPROVED** [review]

File: test/e2e/applicationset_test.go (hunk 1 above)
Comment (wanghong230) [reply test/e2e/applicationset_test.go:919]: Yeah, it is quite confusing here.

Comment (a244591052) [reply test/e2e/applicationset_test.go]: Done

Comment (blakepettersson) [reply test/e2e/applicationset_test.go]: @pasha-codefresh does this look good to you now?

**Pull request merged**